appinventor/components/src/com/google/appinventor/components/runtime/LinearProgress.java

```diff
@@ -0,0 +1,185 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2017-2023 Kodular, All rights reserved
+// Copyright 2023 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package com.google.appinventor.components.runtime;
+
+import android.content.Context;
+import android.graphics.BlendMode;
+import android.graphics.BlendModeColorFilter;
+import android.graphics.PorterDuff;
+import android.graphics.drawable.Drawable;
+import android.os.Build;
+import android.os.Build.VERSION;
+import android.util.Log;
+import android.widget.ProgressBar;
+import com.google.appinventor.components.annotations.*;
+import com.google.appinventor.components.common.ComponentCategory;
+import com.google.appinventor.components.common.PropertyTypeConstants;
+import com.google.appinventor.components.common.YaVersion;
+
+
+@DesignerComponent(
+    version = YaVersion.LINEAR_PROGRESS_COMPONENT_VERSION,
+    category = ComponentCategory.USERINTERFACE,
+    description = "A visible component that indicates the progress of an operation using an animated linear bar.",
+    iconName = "images/linearProgress.png"
+)
+@SimpleObject
+
+public final class LinearProgress extends AndroidViewComponent {
+  private static final String LOG_TAG = "LinearProgress";
+  private Context context;
+  private ProgressBar progressBar;
+  private int progressColor = Component.COLOR_BLUE;
+  private int indeterminateColor = Component.COLOR_BLUE;
+
+  public LinearProgress(ComponentContainer container) {
+    super(container);
+    context = container.$context();
+
+    progressBar = new ProgressBar(context, null, android.R.attr.progressBarStyleHorizontal);
+
+    container.$add(this);
+
+    Minimum(0);
+    Maximum(100);
+    ProgressColor(Component.COLOR_BLUE);
+    IndeterminateColor(Component.COLOR_BLUE);
+    Indeterminate(true);
+    Width(LENGTH_FILL_PARENT);
+    Log.d(LOG_TAG, "Linear Progress created");
+  }
+
+  @Override
+  public ProgressBar getView() {
+    return progressBar;
+  }
+
+  @Override
+  public int Height() {
+    return getView().getHeight();
+  }
+
+  @Override
+  public void Height(int height) {
+    container.setChildHeight(this, height);
+  }
+
+  @Override
+  public void HeightPercent(int height) {
+  }
+
+  @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_INTEGER, defaultValue = "0")
+  @SimpleProperty(description = "Set the lower range of the progress bar to min. "
+      + "This function works only for devices with API >= 26")
+  public void Minimum(int value) {
```

Comment:
I didn't apply this change as it would create an inconsistency with Kodular's components. If we end up merging it with `Slider`, we can come back and change it later to not divert methods as of now.

Reply:
I think you can update `versioning.js` to rename a property/method name. For example:

https://github.com/mit-cml/appinventor-sources/blob/17b754f9cafae1246a7cdd564189212843de46a6/appinventor/blocklyeditor/src/versioning.js#L3088